OpenAI/Packages/com.openai.unity/Runtime/Threads/ThreadsEndpoint.cs

```diff
@@ -345,13 +352,158 @@ public async Task<RunStepResponse> RetrieveRunStepAsync(string threadId, string
         /// <param name="runId">The id of the run to cancel.</param>
         /// <param name="cancellationToken">Optional, <see cref="CancellationToken"/>.</param>
         /// <returns><see cref="RunResponse"/>.</returns>
-        public async Task<RunResponse> CancelRunAsync(string threadId, string runId, CancellationToken cancellationToken = default)
+        public async Task<bool> CancelRunAsync(string threadId, string runId, CancellationToken cancellationToken = default)
         {
             var response = await Rest.PostAsync(GetUrl($"/{threadId}/runs/{runId}/cancel"), string.Empty, new RestParameters(client.DefaultRequestHeaders), cancellationToken);
             response.Validate(EnableDebug);
-            return response.Deserialize<RunResponse>(client);
+            var run = response.Deserialize<RunResponse>(client);
+
+            if (run.Status < RunStatus.Cancelling)
+            {
+                try
+                {
+                    run = await run.WaitForStatusChangeAsync(cancellationToken: cancellationToken);
+                }
+                catch (Exception)
+                {
+                    // ignored
+                }
+            }
+
+            return run.Status >= RunStatus.Cancelling;
```

Comment:
Very helpful small update! Helped me simplify the usage of this. Thanks.

Reply:
@StephenHodgson I just revisited the documentation on the [Run Lifecycle](https://platform.openai.com/docs/assistants/how-it-works/runs-and-run-steps) and I believe the above code would fail in the rare cases of an `Incomplete` Run, because the Enum has it listed before Cancelling.

My quick suggestion would be to simply swap the order in the enum below, but I may be overlooking something there :)

```
public enum RunStatus{ 
...
Cancelling
Incomplete
...
}
```

Reply:
Thanks. I'll address in hotfix in a few days

Reply:
Good idea. That should work.